binaries/runtime/src/operator/shared_lib.rs

```diff
@@ -150,7 +150,10 @@ impl<'lib> SharedLibraryOperator<'lib> {
             DoraResult { error }
         });
 
-        while let Ok(input) = self.inputs.recv() {
+        let reason = loop {
+            let Ok(input) = self.inputs.recv() else {
```

Comment:
Nice :) #Rust 1.65

Reply:
Yeah, I was really looking forward to this feature!